_posts/2016-08-09-the-tech-behind-live-auction-integration.markdown

```diff
@@ -0,0 +1,57 @@
+
+This needs to happen in a tight loop to allow online bidders to be competitive with those in the room and on the phone, and the architecture and UX of the LAI product were optimized for that result. In addition, because we are integrating with live events outside of our direct control, there are many, many ways things can deviate from this idealized flow. We had to carefully account for these situations.
+
+Where possible, we wanted to leverage our existing auctions technology. But in some places, there were opportunities to upgrade that technology and other places where new approaches were necessary. Meanwhile, we were running the busiest Artsy auction season to date, and we had to ensure that we weren’t disrupting our existing stack. Below, I discuss the pieces of the end-to-end product.
+
+# The Live user experience
+
+When the live auction actually begins, participants and Artsy staff interact with the system with front-end software developed from scratch. Web users (desktop and mobile) and staff use a new, dedicated Artsy Live web application, which is implemented in a project we call Prediction. iOS Artsy App users can also participate with newly developed UX within that app.
+
+## The web app: Prediction
+
+Our bidder and staff web interface is implemented in an application we call Prediction, a [universal](https://medium.com/@mjackson/universal-javascript-4761051b7ae9#.ev1yd3juy) [React](https://facebook.github.io/react/)+[Redux](http://redux.js.org/) Javascript app, served from an [Express](http://expressjs.com/) server. Using React allowed us to completely share our view layer code for prerendering in the server and making updates in the client. Famously, it encourages developers to make their views operate like functions. Data are passed in as parameters and the result is a description of the desired output, and the React framework takes charge of updating the user’s view in the browser efficiently.
+
+As our participants and staff are interacting with live views of the same underlying auction state, keeping our state management and transition code organized with Redux allowed us to achieve a massive amount of reuse of model and controller code between our desktop bidding, mobile bidding, and admin interfaces. To solve both Redux's [async](http://stackoverflow.com/q/34570758/807674) and [data conveyance](http://stackoverflow.com/q/34299460/807674) “problems", we built an integration layer for React and Redux called [React Redux Controller](https://github.com/artsy/react-redux-controller). We found the React+Redux approach to model-view-controller app development to be a major win in what it gave us for maintainability, code reuse, easy testability, and the ability to reason about our code.
```

Comment:
Could be worth separating the last sentence out into its own paragraph to drive the point home.